stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -196,14 +154,20 @@ public ResolveNodeResponse resolveNode(JsonRpcPeer peer, JSONObject params)
   public void setAttributesAsText(JsonRpcPeer peer, JSONObject params) {
     final SetAttributesAsTextRequest request = mObjectMapper.convertValue(
         params, SetAttributesAsTextRequest.class);
-    final Object object = mObjectIdMapper.getObjectForId(request.nodeId);
 
-    mDOMProvider.postAndWait(new Runnable() {
-      @Override
-      public void run() {
-        mDOMProvider.setAttributesAsText(object, request.text);
-      }
-    });
+    final Object element = mDocument.getElementForNodeId(request.nodeId);
```

Comment:
Before, `DomProvider#setAttributesAsText(...)` just fell through to `NodeDescriptor#setAttributesAsText(...)`. Now we just set it through an instance of `NodeDescriptor` because that way there is only one place where we can set attributes.